src/main/native/windows/util.h

```diff
@@ -30,18 +30,20 @@ using std::wstring;
 // closed with FindClose (otherwise they aren't closed properly).
 class AutoHandle {
  public:
-  AutoHandle(HANDLE handle = INVALID_HANDLE_VALUE) : handle_(handle) {}
+  explicit AutoHandle(HANDLE handle = INVALID_HANDLE_VALUE) : handle_(handle) {}
+  AutoHandle(const AutoHandle&) = delete;
+  AutoHandle(AutoHandle&& other) = delete;
+  AutoHandle& operator=(const AutoHandle&) = delete;
+  AutoHandle& operator=(AutoHandle&& other) = delete;
 
   ~AutoHandle() {
     if (IsValid()) {
       ::CloseHandle(handle_);
     }
   }
 
-  HANDLE Release() {
-    HANDLE h = handle_;
-    handle_ = INVALID_HANDLE_VALUE;
-    return h;
+  AutoHandle(AutoHandle* other) : handle_(other->handle_) {
+    other->handle_ = INVALID_HANDLE_VALUE;
```

Comment:
Do we have to set `other->handle_` to `INVALID_HANDLE_VALUE` ?

Reply:
Yes. Otherwise both AutoHandles would own the handle and would try to close it.

Reply:
I see.